scripts/cita.sh

```diff
@@ -348,108 +345,117 @@ node_up_check() {
-
-            # Make sure the RabbitMQ fresh
-            clear_rabbit_mq
-
-            do_start "$3" "$4"
-            ;;
-
-        stop)
-            node_up_check
-            do_stop
-            ;;
-
-        restart)
-            node_up_check
-            do_stop
-
-            # Make sure the RabbitMQ fresh
-            clear_rabbit_mq
-
-            do_start "$3" "$4"
-            ;;
-
-        ping)
-            do_ping
-            if [ "${PING_STATUS}" == "pong" ]; then
-                echo "pong"
-            else
-                echo "Node '${NODE_NAME}' not responding to pings."
-                exit 1
-            fi
-            ;;
-
-        top)
-            node_up_check
-            do_top
-            ;;
-        # deprecated, use 'top' instead
-        stat)
-            node_up_check
-            do_top
-            ;;
-        # similar to 'top', but ... ?
-        status)
-            do_status
-            ;;
-
-        logrotate)
-            do_logrotate
-            ;;
-
-        logs)
-            do_logs "$3"
-            ;;
-
-        backup)
-            node_down_check
-            do_backup
-            ;;
-
-        clean)
-            node_down_check
-            do_clean
-            ;;
-
-        *)
-            usage
-            ;;
+    append)
+        config "$@"
+        exit 0
+        ;;
+
+    setup)
+        do_setup
+        ;;
+
+    start)
+        # TODO: should not do so, but present tests need this
+        do_stop
+        node_down_check
+
+        # Make sure the RabbitMQ fresh
+        clear_rabbit_mq
+
+        do_start "$3" "$4"
+        ;;
+
+    stop)
+        node_up_check
+        do_stop
+        ;;
+
+    restart)
+        node_up_check
+        do_stop
+
+        # Make sure the RabbitMQ fresh
+        clear_rabbit_mq
+
+        do_start "$3" "$4"
+        ;;
+
+    ping)
+        do_ping
+        if [ "${PING_STATUS}" == "pong" ]; then
+            echo "pong"
+        else
+            echo "Node '${NODE_NAME}' not responding to pings."
+            exit 1
+        fi
+        ;;
+
+    top)
+        node_up_check
+        do_top
+        ;;
+    # deprecated, use 'top' instead
+    stat)
+        node_up_check
+        do_top
+        ;;
+    # similar to 'top', but ... ?
+    status)
+        do_status
+        ;;
+
+    logrotate)
+        do_logrotate
+        ;;
+
+    logs)
+        do_logs "$3"
+        ;;
+
+    backup)
+        node_down_check
+        do_backup
+        ;;
+
+    clean)
+        node_down_check
+        do_clean
+        ;;
+
+    *)
+        usage
+        ;;
 
     esac
 }
 
+# Test CITA is running in docker.
+check_in_docker() {
+    if grep docker /proc/1/cgroup -qa; then
+        echo "cita_in_docker"
+    else
+        echo "cita_in_host"
+    fi
+}
+
 main() {
-    if [ "$1" != "bebop" ]; then
-        if stat "$CITA_BIN"/cita-env > /dev/null 2>&1; then
+    if [[ "$1" != "bebop" && "cita_in_host" == $(check_in_docker) ]]; then
```

Comment:
Why `check in docker`?

Reply:
We can remove the sub-command `bebop` in a docker environment.

Reply:
But how do you know it's in docker environment before it start?

Reply:
done!